File: Python/easy/118. Pascal's Triangle.py
"""
Given a non-negative integer numRows, generate the first numRows of Pascal's triangle.
"""
class Solution:
    def generate(self, numRows: 'int') -> 'List[List[int]]':
        rList = []
        for i in range(1, numRows + 1):
            if not rList:
                rList.append([1])
            else:
                curList = []
                for j in range(i):
                    if j == 0:
                        curList.append(rList[i - 2][0])
                    elif j == i - 1:
                        curList.append(rList[i - 2][i - 2])
                    else:
                        curList.append(rList[i - 2][j- 1] + rList[i - 2][j])
                rList.append(curList)
        return rList

File: Java/easy/118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        ArrayList<Integer> newRow = new ArrayList<Integer>();
        for(int i = 0; i < numRows; i++)
        {
            newRow.add(0, 1);
            for(int j = 1;j < newRow.size() - 1; j++)
                newRow.set(j, newRow.get(j) + newRow.get(j+1));
            result.add(new ArrayList<Integer>(newRow));
        }
        return result;
    }
}